sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -355,11 +355,12 @@ def is_typing_generic(type_param):
 
   Such objects are considered valid type parameters.
 
-  Always returns false for Python versions below 3.7.
+  For Python versions 3.9 and above, also permits types.GenericAlias.
   """
-  if hasattr(typing, '_GenericAlias'):
-    return isinstance(type_param, typing._GenericAlias)
-  return False
+  if hasattr(types, "GenericAlias") and isinstance(type_param,
+                                                   types.GenericAlias):
+    return True
+  return isinstance(type_param, typing._GenericAlias)
```

Comment:
is typing._GenericAlias always defined for all supported python versions? I worry that we might get attribute errors if the first condition doesn't pass and `typing._GenericAlias` becomes deprecated in some (newer?) python version.. 

Also do we have test coverage for scenario where is_typing_generic returns false ?

Reply:
I believe typing._GenericAlias will always have a definition at this point. Once we drop support for 3.8 we won't have to check for `types.GenericAlias`'s existence either. I added some explicit unit test cases to typehints_test.py just to double-check that we had some coverage